lib/Platform/ElasticSearchPlatform.php

```diff
@@ -389,11 +391,14 @@ private function cleanHost(string $host): string {
 	private function connectToElastic(array $hosts) {
 
 		try {
-			$hosts = array_map([$this, 'cleanHost'], $hosts);
+			$urlResult = $this->processHostUrls($hosts);
+			$hosts = $urlResult['hosts'];
 			$cb = ClientBuilder::create()
 							   ->setHosts($hosts)
 							   ->setRetries(3);
-
+			if ($urlResult['user'] && $urlResult['pass']) {
+				$cb->setBasicAuthentication($urlResult['user'], $urlResult['pass']);
+			}
```

Comment:
- This won't work if there is only a user or only a password. In HTTP Basic Auth providing only one of them is valid.
- You don't need to do these checks as `setBasicAuthentication()` accepts `null` values just fine (they are ignored).

Reply:
Confirm, the same check is already done inside of the client here: https://github.com/elastic/elasticsearch-php/blob/b3af4309bc73fa3b2a1e19f04f9f201916c11417/src/ClientBuilder.php#L379

Thanks, will adjust that.